### PR TITLE
Travis/active middleware

### PR DESCRIPTION
**Pull request by traversng**
@siwan-cb need our help with our flags

### CODE REVIEW

File: src/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+
+// This function can be marked `async` if using `await` inside
+export function middleware(request: NextRequest) {
+  const startDate = new Date('2021-09-01T00:00:00.000Z');
+  const endDate = new Date('2022-09-30T23:59:59.999Z');
```
Comment (siwan-cb) [inline src/middleware.ts:7]: these should be an env variable as well. Look at my previous PR

File: src/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  const now = new Date();
+  const killSwitch = process.env.KILL_SWITCH ?? false;
```
Comment (siwan-cb) [inline src/middleware.ts:9]:
```suggestion
  const killSwitch = Boolean(process.env.KILL_SWITCH);
```

File: src/middleware.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  if (now < startDate || now > endDate || killSwitch) {
```
Comment (siwan-cb) [inline src/middleware.ts:11]: isAfter and isBefore from date-fns are great

File: src/middleware.ts (hunk 1 above)
Comment (siwan-cb) [inline src/middleware.ts:7]: this is already past

File: src/middleware.ts (hunk 2 above)
Comment (siwan-cb) [reply src/middleware.ts:9]: KILL_SWITCH is added on vercel

File: src/middleware.ts (hunk 1 above)
Comment (traversng) [reply src/middleware.ts:7]: this was to test

File: src/middleware.ts (hunk 3 above)
Comment (traversng) [reply src/middleware.ts]: used!

File: src/middleware.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+import { isAfter, isBefore } from 'date-fns';
+
+// This function can be marked `async` if using `await` inside
+export function middleware(request: NextRequest) {
+  const startDate = new Date(process.env.START_DATE as string);
+  const endDate = new Date(process.env.END_DATE as string);
+  const now = new Date();
+  const killSwitch = Boolean(process.env.KILL_SWITCH);
```
Comment (siwan-cb) [inline src/middleware.ts:10]: nice

File: src/middleware.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  if (killSwitch || isBefore(now, startDate) || isAfter(now, endDate)) {
+    return NextResponse.redirect(new URL('/', request.url));
+  }
+}
+
+export const config = {
+  matcher: [
```
Comment (siwan-cb) [inline src/middleware.ts:18]: I think we also need '/art-reveal-screen/:path*`

Comment (traversng) [reply src/middleware.ts:18]: ok, missed this

**Pull request merged**